services/positions-api/src/positions/post.rs

```diff
@@ -35,12 +35,12 @@ impl Service for PositionsPostHandler {
     type Error = hyper::Error;
     type Future = FutureHandled;
 
-    fn call(&self, req: Request) -> Self::Future {
+    fn call(&self, mut req: Request) -> Self::Future {
         let db_conn = self.db_conn.clone();
 
         // Absence of UserID header is definitely an internal server error
-        let user_id = match req.headers().get::<UserID>() {
-            Some(user_id) => user_id.0.clone(),
+        let user_id = match req.headers_mut().remove::<UserID>() {
```

Comment:
What is happening here and why do we remove?

Reply:
Headers::get() returned an reference to the header, so to have it's contents owned we would clone it.

Headers::remove() retrieves an owned Header from internal BTreeMap, hence we do not clone it's contents to pass them wherever we would want.
Since the `Request` and it's headers are to die here anyway, it's safe to take their valuables ;)